core/tasks/handler/htasks/channel_event.go

```diff
@@ -41,66 +40,61 @@ func (t *ChannelEventTask) Type() string {
 	return TypeChannelEvent
 }
 
-func (t *ChannelEventTask) Perform(ctx context.Context, rt *runtime.Runtime, oa *models.OrgAssets, contactID models.ContactID) error {
-	_, err := t.handle(ctx, rt, oa, contactID, nil)
+func (t *ChannelEventTask) UseReadOnly() bool {
+	return !t.NewContact
+}
+
+func (t *ChannelEventTask) Perform(ctx context.Context, rt *runtime.Runtime, oa *models.OrgAssets, contact *models.Contact) error {
+	_, err := t.handle(ctx, rt, oa, contact, nil)
 	return err
 }
 
 // Handle let's us reuse this task's code for handling incoming calls.. which we need to perform inline in the IVR web
 // handler rather than as a queued task.
-func (t *ChannelEventTask) Handle(ctx context.Context, rt *runtime.Runtime, oa *models.OrgAssets, contactID models.ContactID, call *models.Call) (*models.Session, error) {
-	return t.handle(ctx, rt, oa, contactID, call)
+func (t *ChannelEventTask) Handle(ctx context.Context, rt *runtime.Runtime, oa *models.OrgAssets, contact *models.Contact, call *models.Call) (*models.Session, error) {
+	return t.handle(ctx, rt, oa, contact, call)
 }
 
-func (t *ChannelEventTask) handle(ctx context.Context, rt *runtime.Runtime, oa *models.OrgAssets, contactID models.ContactID, call *models.Call) (*models.Session, error) {
-	channel := oa.ChannelByID(t.ChannelID)
-	if channel == nil {
-		slog.Info("ignoring event, couldn't find channel", "channel_id", t.ChannelID)
+func (t *ChannelEventTask) handle(ctx context.Context, rt *runtime.Runtime, oa *models.OrgAssets, contact *models.Contact, call *models.Call) (*models.Session, error) {
+	// if contact is blocked, ignore event
+	if contact.Status() == models.ContactStatusBlocked {
 		return nil, nil
 	}
 
-	// load our contact
-	modelContact, err := models.LoadContact(ctx, rt.ReadonlyDB, oa, contactID)
```

Comment:
It's possible that using `ReadonlyDB` here causes us to miss some events for new contacts because they're not in the read replica yet.. so now we consider `NewContact` on the task payload